### PR TITLE
chore(docs): simplify using-gatsby-image doc

### PR DESCRIPTION
**Pull request by lstar19**
## Description

Fix path for gatsby-config, include entire gatsby-config file structure

## Related Issues

Addresses #11329

### CODE REVIEW

File: docs/docs/using-gatsby-image.md [hunk 1]
```diff
@@ -37,25 +37,68 @@ With Gatsby, we can make images way way better.
...
 
-1. Import gatsby-image and use it in place of the built-in img
-1. Write a GraphQL query using one of the included GraphQL “fragments” which specify the fields needed by `gatsby-image`.
+1. Install `gatsby-image` and its peer dependencies. The `gatsby-config.json` needs to include the following:
+
+```js:title=src/gatsby-config.js
+module.exports = {
+  plugins: [
+    // highlight-start
+    `gatsby-image`,
```
Comment (DSchau) [inline docs/docs/using-gatsby-image.md]: Not necessary to add gatsby-image here!

```suggestion
```

File: docs/docs/using-gatsby-image.md [hunk 2]
```diff
@@ -37,25 +37,68 @@ With Gatsby, we can make images way way better.
...
 export default ({ data }) => (
   <div>
     <h1>Hello gatsby-image</h1>
-    <Img resolutions={data.file.childImageSharp.resolutions} />
+    <Img fluid={data.file.childImageSharp.fluid} />
   </div>
 )
 ```
 
+The GraphQL query creates multiple thumbnails with optimized JPEG and PNG compression. The `gatsby-image` component automatically sets up the “blur-up” effect as well as lazy loading of images further down the screen.
```
Comment (DSchau) [inline docs/docs/using-gatsby-image.md]:
```suggestion
This GraphQL query creates multiple sizes of the image. The image that is appropriate for the current screen resolution (e.g. desktop, mobile, and everything in between). The `gatsby-image` component automatically enables a blur-up effect as well as lazy loading images that are not currently on screen.
```

File: docs/docs/using-gatsby-image.md [hunk 3]
```diff
@@ -37,25 +37,68 @@ With Gatsby, we can make images way way better.
 
 `gatsby-image` is designed to work seamlessly with Gatsby’s native image processing capabilities powered by GraphQL and Sharp. To produce perfect images, you only need to:
 
-1. Import gatsby-image and use it in place of the built-in img
-1. Write a GraphQL query using one of the included GraphQL “fragments” which specify the fields needed by `gatsby-image`.
+1. Install `gatsby-image` and its peer dependencies. The `gatsby-config.json` needs to include the following:
```
Comment (DSchau) [inline docs/docs/using-gatsby-image.md]:
```suggestion
1. Install `gatsby-image` and other, necessary dependencies like `gatsby-plugin-sharp` and `gatsby-transformer-sharp`
    ```shell
    npm i gatsby-image gatsby-transformer-sharp gatsby-plugin-sharp
    ```
1. Add the newly installed plugins and transformer plugins to your `gatsby-config.js`
    ```js:title=gatsby-config.js
    module.exports = {
      plugins: [
        // highlight-start
        `gatsby-plugin-sharp`,
        `gatsby-transformer-sharp`
      ]
    }
    ```
```

File: docs/docs/using-gatsby-image.md (hunk 1 above)
Comment (DSchau) [reply docs/docs/using-gatsby-image.md]: Actually - let's just strike this whole section! It's encapsulated in the suggestion above ☝️

File: docs/docs/using-gatsby-image.md [hunk 4]
```diff
@@ -37,25 +37,68 @@ With Gatsby, we can make images way way better.
...
+    `gatsby-image`,
+    `gatsby-transformer-sharp`,
+    `gatsby-plugin-sharp`,
+    // highlight-end
+  ],
+}
+```
 
-The GraphQL query creates multiple thumbnails with optimized JPEG and PNG compression. The `gatsby-image` component automatically sets up the “blur-up” effect as well as lazy loading of images further down the screen.
+2. Configure the peer dependencies to load images from a folder. In order to use GraphQL to query the image files, the files need to be in a location that is known to Gatsby. This requires another update to `gatsby-config.json`. Any path can replace `src/data` in this example, but whatever the path is the image file being queried needs to be placed within the scope of this path.
```
Comment (DSchau) [inline docs/docs/using-gatsby-image.md]:
```suggestion
2. Configure `gatsby-source-filesystem` to load images from a folder. In order to use GraphQL to query the image files, the files need to be in a location that is known to Gatsby. This requires an to `gatsby-config.js` to configure the plugin. Feel free to replace the `path` option with wherever your images are located relative to your project.
```

File: docs/docs/using-gatsby-image.md [hunk 5]
```diff
@@ -37,25 +37,68 @@ With Gatsby, we can make images way way better.
...
+        path: `${__dirname}/src/data/`,
+      }
+    }
+    // highlight-end
+  ],
+}
+```
 
-Here’s what a really simple Gatsby page component using gatsby-image would look like:
+3. Write a GraphQL query using one of the included GraphQL “fragments” which specify the fields needed by `gatsby-image`. There are numerous fragments that can be used and the choice is based on the type of responsive image desired. This example will use `GatsbyImageSharpFluid`. An example of a GraphQL query is below where the path listed is the path relative to the location specified in the `gatsby-source-filesystem` configuration.
```
Comment (DSchau) [inline docs/docs/using-gatsby-image.md]:
```suggestion
3. Write a GraphQL query using one of the included [GraphQL “fragments”](/packages/gatsby-image/#fragments) which specify the fields needed by `gatsby-image` to create a responsive, optimized image. This example will use `GatsbyImageSharpFluid`. An example of a GraphQL query is below where the path listed is the path relative to the location specified in the `gatsby-source-filesystem` options.
```

**Comment by lstar19** [comment]
I'm good with those changes. I accepted them directly so we'll want to double check the entire file is coherent.

**Comment by lstar19** [comment]
It looks like the modules code snippet now appears twice but I'm not quite sure how to access the most recent version since it's no longer on my forked repo.

File: docs/docs/using-gatsby-image.md [hunk 6]
```diff
@@ -37,25 +37,67 @@ With Gatsby, we can make images way way better.
...
+2. Add the newly installed plugins and transformer plugins to your `gatsby-config.js`
 
-Here’s what a really simple Gatsby page component using gatsby-image would look like:
+```js:title=gatsby-config.js
+module.exports = {
+  plugins: [`gatsby-plugin-sharp`, `gatsby-transformer-sharp`],
+}
+```
+
+3. Configure `gatsby-source-filesystem` to load images from a folder. In order to use GraphQL to query the image files, the files need to be in a location that is known to Gatsby. This requires an to `gatsby-config.js` to configure the plugin. Feel free to replace the `path` option with wherever your images are located relative to your project.
```
Comment (marcysutton) [inline docs/docs/using-gatsby-image.md]: >This requires an to `gatsby-config.js`...

I think this needs a word like `entry` or `edit` here, i.e. "This requires an edit to `gatsby-config.js...`"

File: docs/docs/using-gatsby-image.md [hunk 7]
```diff
@@ -37,25 +37,67 @@ With Gatsby, we can make images way way better.
...
 
 ```jsx
-import React from "react"
 import Img from "gatsby-image"
 
 export default ({ data }) => (
   <div>
     <h1>Hello gatsby-image</h1>
-    <Img resolutions={data.file.childImageSharp.resolutions} />
+    <Img fluid={data.file.childImageSharp.fluid} />
```
Comment (marcysutton) [inline docs/docs/using-gatsby-image.md]: Please add `alt` text here to set a good example for accessibility. You can get pretty creative with it! 😃

`JAM stack jammies`

File: docs/docs/using-gatsby-image.md [hunk 8]
```diff
@@ -37,25 +37,67 @@ With Gatsby, we can make images way way better.
...
   </div>
 )
 ```
 
+This GraphQL query creates multiple sizes of the image. The image that is appropriate for the current screen resolution (e.g. desktop, mobile, and everything in between). The `gatsby-image` component automatically enables a blur-up effect as well as lazy loading images that are not currently on screen.
```
Comment (marcysutton) [inline docs/docs/using-gatsby-image.md]: The second sentence here isn't quite a full sentence. Here's an easy fix: "Gatsby will load the image that is appropriate for the current screen resolution (e.g. desktop, mobile, and everything in between)."

**Comment by lstar19** [comment]
Think this should be ready for review. Not entirely sure why the e2e test failed this last run.

**Comment by DSchau** [comment]
@lstar19 looks like your origin/upstream were a little out of date, so I merge upstream/master into this branch for you. For more context, generally you can fix with:

```
git fetch upstream master
git merge upstream/master
git push origin master # or whatever branch name
```

**Comment by lstar19** [comment]
Thanks! That makes sense, just wish the error message had been a bit more clear.

**Comment by lstar19** [comment]
Should be good to go :)

**Pull request merged**